tests/test-backend-ops.cpp

```diff
@@ -175,6 +175,37 @@ static void init_tensor_kq_mask(ggml_tensor * tensor, float min = -1.0f, float m
     ggml_backend_tensor_set(tensor, data_f16.data(), 0, data_f16.size()*sizeof(ggml_fp16_t));
 }
 
+// generate a lower triangular matrix
+static void init_tensor_tril(ggml_tensor * tensor, float min = -1.0f, float max = 1.0f) {
+    GGML_ASSERT(tensor->type == GGML_TYPE_F32);
+    GGML_ASSERT(tensor->ne[0] == tensor->ne[1]);
+
+    GGML_TENSOR_LOCALS(int32_t, ne, tensor, ne);
+
+    std::vector<float> data_f32(ne0*ne1*ne2*ne3);
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<float> dis(min, max);
+
+    for (int64_t i3 = 0; i3 < tensor->ne[3]; i3++) {
+        for (int64_t i2 = 0; i2 < tensor->ne[2]; i2++) {
+            for (int64_t i1 = 0; i1 < tensor->ne[1]; i1++) {
+                for (int64_t i0 = 0; i0 < tensor->ne[0]; i0++) {
+                    int64_t idx = (i0 * tensor->nb[0] + i1 * tensor->nb[1] + i2 * tensor->nb[2] + i3 * tensor->nb[3]) / sizeof(float);
+                    if (i0 <= i1) {
+                        data_f32[idx] = dis(gen);
+                    } else {
+                        data_f32[idx] = 0.0f;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Not a big problem, but since this function is already using `GGML_TENSOR_LOCALS`, it should use the locals `ne0` `ne1` etc isntead of `tensor->ne`.